torchrl/envs/libs/smacv2.py

```diff
@@ -377,10 +377,9 @@ def _step(self, tensordict: TensorDictBase) -> TensorDictBase:
         obs = self.get_obs()
         state = self.get_state()
         info = self.observation_spec["info"].encode(info)
-        if "episode_limit" not in info.keys():
-            info["episode_limit"] = self.observation_spec["info"][
-                "episode_limit"
-            ].zero()
+        for info_key in self.observation_spec["info"].keys():
+            if info_key not in info.keys():
+                info[info_key] = self.observation_spec["info"][info_key].zero()
```

Comment:
to avoid multiple access to the keys:
```suggestion
        for info_key, spec in self.observation_spec["info"].keys():
            if info_key not in info.keys():
                info[info_key] = spec[info_key].zero()
```